tests/AdaptiveExpressions.Tests/ExpressionParserTests.cs

```diff
@@ -789,7 +789,24 @@ public class ExpressionParserTests
             Test("ticks(timestampObj3)", 636503904000000000),
             Test("ticksToDays(2193385800000000)", 2538.64097222),
             Test("ticksToHours(2193385800000000)", 60927.383333333331),
-            Test("ticksToMinutes(2193385811100000)", 3655643.0185), 
+            Test("ticksToMinutes(2193385811100000)", 3655643.0185),
+            Test("isMatch(getPreviousViableDate('XXXX-07-10'), '20[0-9]{2}-07-10')", true),
```

Comment:
>'20[0-9]{2}-07-10' [](start = 63, length = 18)

These tests are not very precise.  If you don't pass in a base date time, then you can calculate wrt to the current customer datetime/timezone.  You can also pass in the base date time which makes these a precise test.

Reply:
add test cases for timezone

Reply:
> added test cases for timezone